File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        rm = collections.Counter(ransomNote)
        mg = collections.Counter(magazine)

        for k, v in rm.items():
            if k not in mg or mg[k] < v:
                return False

        return True

File: 383-ransom-note/383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int>rn(26,0);
        for(auto x:magazine){
            rn[x-'a']++;
        }
        for(auto x:ransomNote){
            rn[x-'a']--;
            if(rn[x-'a'] < 0 ){
                return false;
            }
        }
        return true;
       
    }
};